kanidm_unix_int/tests/cache_layer_test.rs

```diff
@@ -514,31 +514,58 @@ fn test_cache_account_pam_allowed() {
         let fut = async move {
             cachelayer.attempt_online().await;
 
-            // Should fail
-            let a1 = cachelayer
-                .pam_account_allowed("testaccount1")
-                .await
-                .expect("failed to authenticate");
-            assert!(a1 == Some(false));
+            // ! TRACING BEGIN
+            let span = span!(Level::INFO, "test_cache_account_pam_allowed");
+            let _enter = span.enter();
+            // ? Seems like these subscribers will allow us to have json loggers
+            // ? and human-readable loggers at the same time, and select between
+            // ? them easily?
+
+            let subscriber = FmtSubscriber::builder()
+                .with_max_level(Level::INFO)
+                // .event_format(format::json()) // ! uncomment for JSON
+                .finish();
+            // ? This is specific to only applications-
+            // ? Ask if kanidm will ever be used as a library?
+            // ? I'm guessing not but double check
+
+            // Set a global subscriber (bad)
+            // tracing::subscriber::set_global_default(subscriber)
+            //     .expect("setting default subscriber failed");
+            // ! TRACING END
+
+            // ! This is what I want to do, but I get the error message:
+            // ! `await` is only allowed inside `async` functions and blocks
+            // ! only allowed inside `async` functions and blocks
+            // And if I make the block of the closure async, then there's
+            // nothing that will block on it and actually run it.
```

Comment:
I think maybe you should move the tracing subscriber outside of this section, above `let fut = async move`. That way your setting up tracing in "sync" land, and then that can wrap the `let fut = ...` and `rt.block_on()`. That would resolve your problem I think :)

Reply:
This fixed the issue, thanks!